tests/unit/einstein/test_ceph.py

```diff
@@ -0,0 +1,272 @@
+import time
+import unittest
+import pytest
+import subprocess
+import ims.common.config as config
+
+config.load()
+
+import rados
+import rbd
+import ims.common.constants as constants
+import ims.einstein.ceph as Ceph
+from ims.database.database import Database
+from ims.einstein.operations import BMI
+import ims.exception.file_system_exceptions as file_system_exceptions
+from ims.common.log import trace
+from pytest import set_trace as bp
+import logging
+
+_cfg = config.get()
+
+PROJECT = _cfg.tests.project
+NETWORK = _cfg.tests.network
+
+EXIST_IMG_NAME = _cfg.tests.exist_img_name
+NEW_SNAP_NAME = _cfg.tests.new_snap_name
+NOT_EXIST_IMG_NAME = _cfg.tests.not_exist_img_name
+NOT_EXIST_SNAP_NAME = _cfg.tests.not_exist_snap_name
+
+HIL_CALL_TIMEOUT = 5  # constants.HIL_CALL_TIMEOUT
+
+POOL = _cfg.fs.pool
+# test image
+CEPH_IMG = "BMI_TEST_CEPH"
+# test image size
+CEPH_IMG_SIZE = 1024
+# test snap_image
+CEPH_EXISTING_IMG = "bmi_ci.img"
+CEPH_SNAP_IMG = "BMI_TEST_CEPH_SNAP"
+# test write
+TEST_DATA = 'bmi' * 10
+OFFSET = 0
+# test clone
+CEPH_CHILD_IMG = "BMI_TEST_CEPH_CHILD_IMAGE"
+
+
+class TestListImages(unittest.TestCase):
+    """
+    Test list images
+    """
+    @trace
+    def setUp(self):
+        self.fs = Ceph.RBD(_cfg.fs, _cfg.iscsi.password)
+        time.sleep(HIL_CALL_TIMEOUT)
+
+    def runTest(self):
+        imgs = self.fs.list_images()
+        command = "echo {0} | sudo -S rbd --keyring {1} --id {2} ls " \
+                  "{3}".format(self.fs.password, self.fs.keyring,
+                               self.fs.rid, self.fs.pool)
+        p = subprocess.Popen(command, shell=True, stderr=subprocess.STDOUT,
+                             stdout=subprocess.PIPE)
+        output, err = p.communicate()
+        rbd_imgs = output.split()
+        self.assertListEqual(imgs, rbd_imgs)
```

Comment:
It is better to create an image in setUp and check if it is there as both will be an empty array most likely.